beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -148,9 +162,12 @@ func (vs *Server) GetBeaconBlock(ctx context.Context, req *ethpb.BlockRequest) (
 		// Get local and builder (if enabled) payloads. Set execution data. New in Bellatrix.
 		var overrideBuilder bool
 		var localPayload interfaces.ExecutionData
-		localPayload, blobBundle, overrideBuilder, err = vs.getLocalPayloadAndBlobs(ctx, sBlk.Block(), head)
+		localPayload, blobBundle, overrideBuilder, err := vs.getLocalPayloadAndBlobs(ctx, sBlk.Block(), head)
 		if err != nil {
-			return nil, status.Errorf(codes.Internal, "Could not get local payload: %v", err)
+			log.WithError(err).Warn("failed to retrieve local payload; attempting builder payload if available")
```

Comment:
By allowing local payload to be nil then below there are several constructions that need to check for nil, for example we use `blobBundle` to set `fullBlobs` in line 199. These would have to use the blinded version if local failed, etc.

Reply:
the point is that the line 
```
        fullBlobs, err := blobsBundleToSidecars(blobBundle, sBlk.Block())
```
will return nil (and even no error) if `blobsBundle` is nil, when we should be using the blinded blobs bundle

Reply:
9a5dc4f6c